tests/e2e/nvidia-ctk_containerd_test.go

```diff
@@ -0,0 +1,850 @@
+				}
+
+				// Install the NVIDIA Container Toolkit packages
+				_, _, err = toolkitInstaller.Install(nestedContainerRunner)
+				Expect(err).ToNot(HaveOccurred(), "Failed to install toolkit for containerd")
+			})
+
+			AfterAll(func(ctx context.Context) {
+				// Cleanup: remove the container
+				_, _, err := runner.Run(fmt.Sprintf("docker rm -f %s 2>/dev/null || true", containerName))
+				if err != nil {
+					GinkgoLogr.Error(err, "failed to cleanup container", "container", containerName)
+				}
+			})
+
+			BeforeEach(func(ctx context.Context) {
+				// No setup needed - each test starts with the state from the previous test
+			})
+
+			AfterEach(func(ctx context.Context) {
+				// Step 1: Restore containerd configuration from backup
+				_, _, err := nestedContainerRunner.Run(`
+			# Restore the original conf.d
+			if [ -d /tmp/containerd-conf.d.backup ]; then
+				rm -rf /etc/containerd/conf.d
+				cp -r /tmp/containerd-conf.d.backup /etc/containerd/conf.d
+			else
+				# If no backup exists, just clean up
+				rm -rf /etc/containerd/conf.d
+				mkdir -p /etc/containerd/conf.d
+			fi
+			
+			# Restore the original config.toml
+			if [ -f /tmp/containerd-config.toml.backup ]; then
+				cp /tmp/containerd-config.toml.backup /etc/containerd/config.toml
+			fi
+		`)
+				Expect(err).ToNot(HaveOccurred(), "Failed to restore containerd configuration")
+
+				// Step 2: Restart containerd and wait for it to be ready
+				err = restartContainerdAndWait(nestedContainerRunner)
+				Expect(err).ToNot(HaveOccurred(), "Failed to restart containerd after configuration restore")
+			})
+
+			When("configuring containerd", func() {
+				It("should add NVIDIA runtime using drop-in config without modifying the main config", func(ctx context.Context) {
+					// Configure containerd using nvidia-ctk
+					_, _, err := nestedContainerRunner.Run(`nvidia-ctk runtime configure --runtime=containerd --config=/etc/containerd/config.toml --drop-in-config=/etc/containerd/conf.d/99-nvidia.toml --set-as-default --cdi.enabled`)
+					Expect(err).ToNot(HaveOccurred(), "Failed to configure containerd")
+
+					// restart containerd
+					err = restartContainerdAndWait(nestedContainerRunner)
+					Expect(err).ToNot(HaveOccurred(), "Failed to restart containerd")
+
+					// Verify containerd loaded the config correctly
+					output, _, err := nestedContainerRunner.Run(`containerd config dump`)
+					Expect(err).ToNot(HaveOccurred())
+
+					// Parse the TOML output
+					config, err := parseContainerdConfig(output)
+					Expect(err).ToNot(HaveOccurred(), "Failed to parse containerd config")
+
+					// Verify config version
+					version := config.Get("version")
+					Expect(version).To(Equal(int64(env.configVersion)))
+
+					// Verify imports
+					if env.configVersion == 2 {
+						// containerd 1.7 shows actual resolved imports
+						err = validateImports(config, []string{"/etc/containerd/conf.d/99-nvidia.toml"}, true)
```

Comment:
This is a sideeffect of the `config dump` command and does not reflect the actual content of the config file. One question I do have is why we're using `validateImports` here and not for the non v2 case?